src/solvers/dgmulti/flux_differencing_gauss_sbp.jl

```diff
@@ -41,505 +41,566 @@ abstract type AbstractTensorProductGaussOperator end
-      # This is due to way we define face nodes for Hex() types in StartUpDG.jl.
-      out[index_left]  = out[index_left]  + interp_matrix_gauss_to_face_1d[1, jj] * x[j, i, jj]
-      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[j, i, jj]
+
+    # interpolation in the z-direction
+    @turbo for i in Base.OneTo(nnodes_1d), j in Base.OneTo(nnodes_1d) # loop over nodes in a face
+        index_left = face_indices_tensor_product[1, i, j, 3]
+        index_right = face_indices_tensor_product[2, i, j, 3]
+        for jj in Base.OneTo(nnodes_1d) # loop over "line" of volume nodes
+            # The ordering (i,j) -> (j,i) needs to be reversed for this last face.
+            # This is due to way we define face nodes for Hex() types in StartUpDG.jl.
+            out[index_left] = out[index_left] +
+                              interp_matrix_gauss_to_face_1d[1, jj] * x[j, i, jj]
+            out[index_right] = out[index_right] +
+                               interp_matrix_gauss_to_face_1d[2, jj] * x[j, i, jj]
+        end
     end
-  end
 end
 
 # Projects face node values to volume Gauss nodes on one element.
 @inline function tensor_product_gauss_face_operator!(out_vec::AbstractVector,
-                                                     A::TensorProductGaussFaceOperator{2, Projection{ApplyFaceWeights}},
-                                                     x::AbstractVector) where {ApplyFaceWeights}
-
-  (; interp_matrix_gauss_to_face_1d, face_indices_tensor_product) = A
-  (; inv_volume_weights_1d, nnodes_1d) = A
-
-  fill!(out_vec, zero(eltype(out_vec)))
-
-  # As of Julia 1.9, Base.ReshapedArray does not produce allocations when setting values.
-  # Thus, Base.ReshapedArray should be used if you are setting values in the array.
-  # `reshape` is fine if you are only accessing values.
-  # Note that, for 2D GaussSBP nodes, the indexing is first in x, then y
-  out = Base.ReshapedArray(out_vec, (nnodes_1d, nnodes_1d), ())
-
-  if ApplyFaceWeights == true
-    @turbo for i in eachindex(x)
-      x[i] = x[i] * A.face_weights[i]
+                                                     A::TensorProductGaussFaceOperator{
+                                                                                       2,
+                                                                                       Projection{
+                                                                                                  ApplyFaceWeights
+                                                                                                  }
+                                                                                       },
+                                                     x::AbstractVector) where {
+                                                                               ApplyFaceWeights
+                                                                               }
```

Comment:
Not sure how I feel about this. On one hand, it is consistent, but IMO it makes the function signatures harder to read when type parameters are expanded so much.

Reply:
This is an abomination, I agree. In this case, I recommend to just use
```julia
#! format: off
manually formatted code
#! format: on
```